internal/querynodev2/segments/metricsutil/observer.go

```diff
@@ -0,0 +1,245 @@
+package metricsutil
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/atomic"
+
+	"github.com/milvus-io/milvus/pkg/metrics"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+// labeledRecord is a labeled sample point.
+type labeledRecord interface {
+	// Label of the access metric.
+	Label() SegmentLabel
+
+	// Finish finishes the record.
+	Finish(err error)
+
+	// getError returns the error of the record.
+	// current metric system simply reject the error operation.
+	getError() error
+}
+
+// globalObserver is the global resource groups observer.
+var (
+	once           sync.Once
+	globalObserver *segmentsObserver
+)
+
+func getGlobalObserver() *segmentsObserver {
+	once.Do(func() {
+		globalObserver = newSegmentsObserver()
+		go func() {
+			d := 15 * time.Minute
+			ticker := time.NewTicker(d)
```

Comment:
We'd better make d configurable

Reply:
it's just a metric expire time. we do not modify it frequently.

configurable metric expire time will be given at main PR: #31562